vernac/himsg.ml

```diff
@@ -568,7 +568,11 @@ let rec explain_evar_kind env sigma evk ty = function
       str " in the partial instance " ++ pc ++
       str " found for ") ++
       explain_evar_kind env sigma evk'
-      (pr_leconstr_env env sigma ty') (snd evi.evar_source)
+        (pr_leconstr_env env sigma ty') (snd evi.evar_source)
+  | Evar_kinds.RecordFieldEvar (field, constructor) ->
+          let (name_mut_ind, _) = Names.inductive_of_constructor constructor in
+     str "field " ++ Names.Constant.print field ++ str " of record " ++
+     Names.MutInd.print name_mut_ind
```

Comment:
You are going here too low-level. Just do:
```ocaml
let ind = Names.inductive_of_constructor constructor in
  str "field " ++ Printer.pr_constant env field ++ str " of record " ++ Printer.pr_inductive env ind
```
The printer `Printer.pr_inductive` shall compute the correct name for shortest name for printing and the correct name for mutual inductive types. Similarly, the printer `pr_constant` shall compute the "best" name for a constant to be understandable in the current context.

The printers `Names.MutInd.print` and `Names.constant.print` are low-level and should probably be renamed into `debug_print` (@ppedrot, @maximedenes, @ejgallego, ... what do you think?). It shall print only the key name of a block of inductive types (for instance if you have `Inductive even := ... with odd := ...`, it would not print `odd` but only `even` which is the key name for the block of both types). Otherwise said, internally, an inductive type is represented by a pair `(keyname_of_block,index_in_block)`, and, here you are dropping the index and printing only the key name.